crates/ironrdp-blocking/src/connector.rs

```diff
@@ -125,8 +125,19 @@ where
 {
     assert!(connector.should_perform_credssp());
 
-    let (mut sequence, mut ts_request) =
-        CredsspSequence::init(connector, server_name, server_public_key, kerberos_config)?;
+    let selected_protocol = match connector.state {
+        ClientConnectorState::Credssp { selected_protocol, .. } => selected_protocol,
+        _ => unreachable!(),
```

Comment:
suggestion: Can you replace the `unreachable!` with something like
```rust
return Err(general_err!("invalid connector state for CredSSP sequence"));
```
Like we previously had? This should indeed never happen unless there is a bug, but I prefer not failing with an unrecoverable error. IronRDP is also used in backends and should kill threads.